static/js/compose_validate.js

```diff
@@ -64,6 +65,40 @@ export function needs_subscribe_warning(user_id, stream_id) {
     return true;
 }
 
+export function warn_if_pm_mentions_user(mention_text, full_name, mentioned_user_id) {
+    if (compose_state.get_message_type() !== "private") {
+        return mention_text;
+    }
+    const recipient = compose_state.private_message_recipient();
+    const emails = util.extract_pm_recipients(recipient);
+    const to_user_ids = people.email_list_to_user_ids_string(emails);
+    if (to_user_ids === mentioned_user_id.toString()) {
+        const $warning_area = $("#compose_pm_mentions_user_alert");
+        const context = {full_name};
+        const new_row = render_compose_pm_mentions_user_alert(context);
+        const mention_str = people.get_mention_syntax(full_name, mentioned_user_id, false);
+        const silent_mention_str = people.get_mention_syntax(full_name, mentioned_user_id, true);
+        mention_text = mention_text.replace(mention_str, silent_mention_str);
+        // also replace other mentions...
+        compose_ui.replace_syntax(mention_str, silent_mention_str);
+        $warning_area.data("full-name", full_name);
```

Comment:
This can be removed once we change the code as mentioned in https://github.com/zulip/zulip/pull/20614/files#r841834990.

Reply:
DONE. good catch.